src/app/vendors/VendorItems.tsx

```diff
@@ -86,7 +87,22 @@ export default function VendorItems({
     return <div className={styles.vendorContents}>{t('Vendors.NoItems')}</div>;
   }
 
-  const itemsByCategory = _.groupBy(vendor.items, (item: VendorItem) => item.displayCategoryIndex);
+  // remove deprecated mods from seasonal artifact
+  if (vendor.def.hash === VENDORS.ARTIFACT) {
+    const first = vendor.items.findIndex((firstDepMod) =>
+      deprecatedMods.includes(firstDepMod?.item?.hash ?? 0)
+    );
+    const reverseList = [...vendor.items].reverse();
+    const last =
+      vendor.items.length -
+      2 -
+      reverseList.findIndex((lastDepMod) => deprecatedMods.includes(lastDepMod?.item?.hash ?? 0));
+    if (first > -1 && last > -1) {
+      vendor.items.splice(first, last);
+    }
+  }
+
+  const itemsByCategory = _.groupBy(vendor.items, (item: VendorItem) => item?.displayCategoryIndex);
```

Comment:
```suggestion
  // remove deprecated mods from seasonal artifact
  if (vendor.def.hash === VENDORS.ARTIFACT) {
    vendor.items = vendor.items.filter(
      (i) => i.item?.hash && !deprecatedMods.includes(i.item.hash)
    );
  }

  const itemsByCategory = _.groupBy(vendor.items, (item) => item?.displayCategoryIndex);
```

i'll hand it to to you the splice is clever but i am not sure why it can't be this